addons/account/static/src/js/account_report_widgets.js

```diff
@@ -1,604 +1,648 @@
+                                $("input[name='date_from']").val(dt.toISOString().substr(0, 10));
+                            }
+                        });
                     });
-                    $('#footnoteModal').modal('show');
                 });
-                
-            },
-            editSummary: function(e) {
-                e.stopPropagation();
-                e.preventDefault;
-                var $el = $(e.target);
-                var text = $el.html().replace(/\s+/g, ' ').replace(/\r?\n/g, '').replace(/<br>/g, '\n').replace(/(\n\s*)+$/g, '');
-                $el.replaceWith(openerp.qweb.render("editContent", {num: 0, text: text}));
-            },
-            clickPencil: function(e) {
-                e.stopPropagation();
-                e.preventDefault();
-                if ($(e.target).parents("div.oe-account-summary, p.footnote").length > 0) {
-                    var num = 0;
-                    if ($(e.target).parent().parent().is("p.footnote")) {
-                        $(e.target).parent().parent().attr('class', 'footnoteEdit')
-                        var $el = $(e.target).parent().parent().find('span.text');
-                        var text = $el.html().replace(/\s+/g, ' ').replace(/\r?\n/g, '').replace(/<br>/g, '\n').replace(/(\n\s*)+$/g, '');
-                        text = text.split('.');
-                        var num = text[0];
-                        text = text[1];
-                        $el.html(openerp.qweb.render("editContent", {num: num, text: text}));
+                break;
+        }
+        if (filter == 'custom') {
+            this.$(".custom-date").css("visibility", "visible");
+        }
+        else {
+            this.$(".custom-date").css("visibility", "hidden");
+        }
+        this.onChangeCmpDateFilter();
+    },
+    onChangeCmpDateFilter: function() {
+        if ($(".date_bank_reconciliation").length > 0) return;
+        var date_filter = this.$("select[name='date_filter']").val();
+        var cmp_filter = this.$("select[name='date_filter_cmp']").val();
+        var no_date_range = this.$("input[name='date_from']").length == 0;
+        if (cmp_filter == 'custom') {
+            this.$(".custom-cmp").css("display", "inline");
+            this.$("input[name='periods_number']").parent().attr('style', 'display: none');
+            this.$("input[name='periods_number']").val(1);
+        }
+        else if (cmp_filter == 'no_comparison') {
+            this.$(".custom-cmp").css("display", "none");
+            this.$("input[name='periods_number']").parent().attr('style', 'display: none');
+        }
+        else {
+            this.$(".custom-cmp").css("display", "none");
+            this.$("input[name='periods_number']").parent().attr('style', '');
+            var dtTo = this.$("input[name='date_to']").val(); 
+            dtTo = new Date(dtTo.substr(0, 4), dtTo.substr(5, 2) - 1, dtTo.substr(8, 2), 12, 0, 0, 0);
```

Comment:
won't work if user enter a date in a different format in the date_to input field, should use new moment().toDate() instead and check date validity with openerp.web.parse_value